src/XrdPosix/XrdPosixMap.cc

```diff
@@ -80,6 +80,41 @@ mode_t XrdPosixMap::Flags2Mode(dev_t *rdv, uint32_t flags)
 
    return newflags;
 }
+
+/******************************************************************************/
+/*                             S t a t F i l l                                */
+/******************************************************************************/
+
+int XrdPosixMap::Entry2Buf(const XrdCl::DirectoryList::ListEntry &dirEnt, struct stat &buf)
```

Comment:
As I mentioned where this is called, this method should be allowed to return the reason for the failure as a text message as well. EIO doesn't capture much here. Of course, that assumes the we may have more than one error return and more than one failure condition could occur. I'm not insisting on this as it's unlikely to occur.

Reply:
I refactored this to set the error message object directly, giving a (somewhat) more descriptive error message.